bgbb/bgbb_utils.py

```diff
@@ -0,0 +1,120 @@
+from collections import OrderedDict
+from itertools import count
+from typing import List, Optional, Sequence, Tuple
+
+import numpy as np
+
+# sim_dat = gen_buy_die(n_opps, n_users, abgd=SimPars.lst, p_th=[pp, th])
+import numpy.random as nr
+
+from numba import njit
+from pandas import DataFrame
+
+Prob = float
+
+
+###############
+# Simulations #
+###############
+def _gen_probs(abgd: List[float], n=10):
+    pa, pb, ta, tb = abgd
+    p = nr.beta(pa, pb, size=n)
+    th = nr.beta(ta, tb, size=n)
+    return p, th
+
+
+def gen_buy_die(
+    n_opps,
+    n_users,
+    abgd: List[float],
+    p_th: Optional[Tuple[Sequence[Prob], Sequence[Prob]]] = None,
+    n_opps_name="n_opps",
+    seed=0,
+):
+    """
+    """
+    nr.seed(seed)
+    if p_th is None:
+        raise NotImplementedError
+        # to dict
+        p, th = _gen_probs(abgd, n=n_users)
```

Comment:
This code pathway looks unreachable.

Reply:
Got rid of NotImplementedError and made test for this case